src/queue.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+function Node( data ) {
+  this._data = data
+  this._next = null
+}
+
+export default class Queue {
+  constructor() {
+    this._front = null
+    this._back = null
+    this._length = 0
+  }
+
+  length() {
+    return this._length
+  }
+
+  isEmpty() {
+    return this._length == 0
+  }
+
+  back() {
+    return this._back._data
+  }
+
+  front() {
+    return this._front._data
+  }
+
+  enqueue( data ) {
+    let newNode = new Node( data )
+    let oldBack = this._back
```

Comment:
Declare and initialize `oldBack` in `else` closer to where you're actually using it.

Makes for more readable code when you declare local variables as close as possible to the context in which they're used.

Also, use `const` when you don't plan on reinitializing a variable.